contracts/btc-finality/src/finality.rs

```diff
@@ -615,16 +598,17 @@ pub fn tally_blocks(
     for h in start_height..=env.block.height {
         let mut indexed_block = BLOCKS.load(deps.storage, h)?;
         // Get the finality provider set of this block
-        let fp_set = FP_SET.may_load(deps.storage, h)?;
+        let fp_power_table = get_power_table_at_height(deps.storage, h)?;
+        let has_fp = !fp_power_table.is_empty();
 
-        match (fp_set, indexed_block.finalized) {
-            (Some(fp_set), false) => {
+        match (has_fp, indexed_block.finalized) {
+            (true, false) => {
                 // Has finality providers, non-finalised: tally and try to finalise the block
```

Comment:
@gitferry After this change, tallying is using the voting power table in BTC finality contract. This was also not correct because the "voting power" in BTC staking contract is not correct and does not account for timestamped pub rand or jailing